runtime/near-test-contracts/estimator-contract/src/lib.rs

```diff
@@ -1054,6 +1054,38 @@ pub unsafe fn yield_create_byte_1000b_argument_length() {
     }
 }
 
+#[no_mangle]
+pub unsafe fn yield_resume_base_prepare() {
+    const METHOD_NAME: &str = "n";
+    for i in 0..255u8 {
+        promise_yield_create(METHOD_NAME.len() as u64, METHOD_NAME.as_ptr() as u64, 0, 0, 0, 1, 2);
+        storage_write(1, core::ptr::addr_of!(i) as u64, u64::MAX, 2, 3);
+    }
+}
+
+#[no_mangle]
+pub unsafe fn yield_resume_base() {
+    for i in 0..255u8 {
+        match storage_read(1, core::ptr::addr_of!(i) as u64, 0) {
```

Comment:
Yes, I anticipte that the cost for this extra read will be negligible. As Jakob explained to me, it was never really a goal to perfectly isolate the measured functions, mostly because we largely care about two things:

1. Not undercharging;
2. That the fees are reasonable enough to be usable.

Since these were the goals, the estimator was not architected with the perfect isolation in mind, and most of the estimations do not isolate various costs for they are just not deemed important enough.

If the cost of `resume` does end up being a pain point we can look at isolating it better in the future, but I think 1Tgas is reasonable enough to get us started, especially since this Tgas is expected to be called from a fresh contract call.